accepted/0000-hub_retail.md

```diff
@@ -0,0 +1,213 @@
+- Feature Name: Hub Retail Extension
+- Start Date: 2020-11-30
+- RFC PR: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+This RFC describes an extension of retail features for Hub environment.
+
+# Motivation
+[motivation]: #motivation
+
+Reasons for use of Hub with Retail environment
+----------------------------------------------
+
+- scalability
+  - support large environments with the number of terminal exceeding limits of one SUMA server
+
+- disconnected operation
+  - Peripheral server must stay fully functional when the Hub connection is down
+
+
+Possible configurations
+-----------------------
+
+1. Full
+```
+    Terminal -> Branch Server -> Peripheral Server -> Hub
+```
+
+2. Combined SUMA and Branch server 
+```
+    Terminal -> Branch + Peripheral Server Combined -> Hub
+```
+
+3. Standard
+```
+    Terminal -> Branch Server -> SUMA Server
+```
+
+4. Single server
+```
+    Terminal -> Branch + SUMA Server Combined
+```
+
+
+Global Retail data
+------------------
+
+The following items are typically shared in whole organization, so they should be visible and 
+manageable via Hub.
+
+  - Images - distributed as large tarballs via http
+  - Image metadata - pillars, list of packages
+  - HWTYPE configurations - group formula
+  
+Local data
+----------
+
+The following items do not need central management and can stay on Peripheral Server.
+
+  - Branch groups -  groups containing Branch Server and connected terminals
+  - Branch formulas - describe configuration of Branch services - network, pxe, dhcp, dns
+  - Branch PXE data - entries describing how to boot each terminal
+  
+# Detailed design
+[design]: #detailed-design
+
+Images + Image metadata
+-----------------------
+
+Current situation: an image is built on Buildhost, then pushed to SUMA Server htdocs directory. Then SUMA creates a db entry (including package list) and a pillar file.
+The pillar conains URL of image tarball, checksums and additional deployment details. It is used for image synchronization and deployment. These data are extracted 
+on Buildhost at the end of build phase. Some of the data can't be extracted later form the tarball.
+
+The SUMA API related to images is rather docker-centered. For Kiwi images it allows to list images and packages and delete an image.
+
+Use cases currently not covered:
+
+- Move image tarball to external HTTP server
+  - In large environments it makes sense to use one or more servers for distribution of images, it should not be the Hub itself.
+- Copy image metadata from one SUMA Server to another 
+  - Hub - copy to all servers
+  - Test in lab, then move the same image to production.
+- Adjust deployment details
+
+
+Proposed solution:
+
+- Implement API calls to
+  - Get Kiwi Image pillar
+  - Update Kiwi Image pillar
+  - Export Kiwi Image (complete metadata)
```

Comment:
There is one use case which is not directly related to Hub but Vladimir keeps in mind as well.
We have a user who have multiple SUSE Manager servers. One for dev, second for QA and third for prod. They build image on dev machine and want the very same image to be tested by QA and then moved to production. No modification to image allowed. For this scenario it would be great to have image import/export and this IMO can help with Hub as well.

Imagine in Hub architecture one Uyuni server to be dedicated to Development. When work is done here then using some mechanism import image to Hub server and then we can use whatever mechanism to distribute the images.

Reply:
ISS is designed to move any content from any Server to any Server. Its main use case will be from Hub to peripheral servers, but nothing stops an enterprising user to come up with more elaborated schemes.

This one in particular looks pretty convoluted to me, but I guess we can (and actually should) leave this particular door open.